src/main/kotlin/uk/gov/dwp/dataworks/dks/services/impl/AESCipherService.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.dwp.dataworks.dks.services.impl
+
+import org.apache.commons.compress.compressors.CompressorStreamFactory
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import uk.gov.dwp.dataworks.dks.services.CipherService
+import java.io.BufferedReader
+import java.io.InputStream
+import java.security.Key
+import java.security.Security
+import java.util.*
+import javax.crypto.Cipher
+import javax.crypto.CipherInputStream
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+object AESCipherService : CipherService {
+     const val cipherAlgorithm: String = "AES/CTR/NoPadding"
+
+    init {
+        Security.addProvider(BouncyCastleProvider())
+    }
+
+    override fun decrypt(key: String, initializationVector: String, content: InputStream): String {
```

Comment:
We should match the names of params to the same as the method they are overriding here

Reply:
Not necessariliy as they match by type and order . However its a good convention to have same in both interface and class so changed it accordingly